File: leet-code/kidsWithCandies.js
/**
 * @param {number[]} candies
 * @param {number} extraCandies
 * @return {boolean[]}
 */
var kidsWithCandies = function (candies, extraCandies) {
  if (
    candies.length < 2 ||
    candies.length > 100 ||
    candies[0] < 1 ||
    candies[candies.length - 1] > 100 ||
    extraCandies < 1 ||
    extraCandies > 50
  ) {
    return []
  }
  
  let max = 0
  for (const candy of candies) {
    max = Math.max(max, candy)
  }
  for (let index = 0; index < candies.length; index++) {
    candies[index] = candies[index] + extraCandies >= max
  }
  return candies
}

console.log(kidsWithCandies([1, 2, 3, 4, 5], 2))

File: leet-code/kidsWithCandies.go
package main

func kidsWithCandies(candies []int, extraCandies int) []bool {
	max := candies[0]
	for _, val := range candies {
		if val > max {
			max = val
		}
	}
	result := make([]bool, len(candies))
	for i, val := range candies {
		if val+extraCandies >= max {
			result[i] = true
		} else {
			result[i] = false
		}
	}
	return result
}
